prow/crier/reporters/gcs/reporter.go

```diff
@@ -110,35 +105,25 @@ func (gr *gcsReporter) reportFinishedJob(ctx context.Context, pj *prowv1.ProwJob
 		return fmt.Errorf("failed to marshal finished metadata: %v", err)
 	}
 
-	bucketName, dir, err := util.GetJobDestination(gr.cfg, pj)
-	if err != nil {
-		return fmt.Errorf("failed to get job destination: %v", err)
-	}
-
+	gr.logger.Debugf("Would upload finished.json info to %q/%q", bucket, dir)
 	if gr.dryRun {
-		gr.logger.Infof("Would upload finished.json info to %q/%q", bucketName, dir)
 		return nil
 	}
-	return util.WriteContent(ctx, gr.logger, gr.author, bucketName, path.Join(dir, prowv1.FinishedStatusFile), false, output)
+	return util.WriteContent(ctx, gr.logger, gr.author, bucket, path.Join(dir, prowv1.FinishedStatusFile), false, output)
 }
 
-func (gr *gcsReporter) reportProwjob(ctx context.Context, pj *prowv1.ProwJob) error {
+func (gr *gcsReporter) reportProwjob(ctx context.Context, bucket, dir string, pj *prowv1.ProwJob) error {
```

Comment:
I am willing to be convinced otherwise, but I don't love this change. A given prowjob has a specific place it is supposed to be written to, and having to specify what the correct place is will introduce more room for caller error.

Calls to `GetJobDestination` should not be _too_ heavy — IIRC it's just reading the config (which is does take a read lock and apparently for some reason involve a bunch of copying), followed by some enum switching and string concatenation. Is there some profiling indicating this is too slow, or a motivation to increase the flexibility of that method?

Reply:
The main reason I made this change is the code is duplicated in three different places and looks not that nice. But now I can see the reason for writing it like this.

I have reverted it.